src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -270,29 +269,41 @@ private void paintUnscaledBorder(Component c, Graphics g,
                 highlight = MetalLookAndFeel.getControlShadow();
                 shadow = MetalLookAndFeel.getControlInfo();
             }
+        }
 
-            // scaled border
-            int thickness = (int) Math.ceil(4 * scaleFactor);
-
+        private void paintUnscaledBorder(Component c, Graphics g,
+                                         int width, int height,
+                                         double scaleFactor) {
+            updateColors(c);
             g.setColor(background);
+
+            // scaled thickness
+            int thickness = (int) Math.ceil(4 * scaleFactor);
```

Comment:
```suggestion
            updateColors(c);

            // scaled thickness
            int thickness = (int) Math.ceil(4 * scaleFactor);
            g.setColor(background);
```
Setting background color logically belongs to painting the bulk of the border rather than updating the colors.